### ISSUE TITLE
Sentence Transformer model can not converge

### ISSUE THREAD

**Issue by TPF2017** [UKPLab/sentence-transformers#275]
I noticed that no matter how long I train the SentenceTransformer model, the model cannot overfit the training data. I want to know why?

**Comment by nreimers** [comment]
By default, it uses a learning rate schedular, that decreases the learning rate with every time step. At some point, the LR is so small that nearly no updates will happen. This should also prevent an too extreme overfitting to your training data.

If you disable it, the model will of course memorize the training data and might not generalize to unseen data (i.e. overfit)

**Comment by TPF2017** [comment]
> By default, it uses a learning rate schedular, that decreases the learning rate with every time step. At some point, the LR is so small that nearly no updates will happen. This should also prevent an too extreme overfitting to your training data.
> 
> If you disable it, the model will of course memorize the training data and might not generalize to unseen data (i.e. overfit)

I have tried the warmupconstant scheduler, but it still cannot overfit the small training data set. Isn’t this strange?

**Comment by nreimers** [comment]
How is your setup? How do you measure overfitting?

**Comment by TPF2017** [comment]
I used SoftmaxLoss and BinaryEmbeddingSimilarityEvaluator, when I changed SoftmaxLoss to CosineSimilarityLoss, this porblem soved. I think because there are some additional parameters in SoftmaxLoss which I didn't consider about it.
